met-web/src/components/engagement/dashboard/comment/EngagementUserComments.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useContext } from 'react';
+import { Button, Grid } from '@mui/material';
+import { useAppSelector } from 'hooks';
+import { Link } from 'react-router-dom';
+import { useNavigate } from 'react-router';
+import { ActionContext } from '../../view/ActionContext';
+import { EngagementBanner } from '../../view/EngagementBanner';
+import { PreviewBanner } from '../../view/PreviewBanner';
+import { SubmissionStatus } from 'constants/engagementStatus';
+import { AppConfig } from 'config';
+
+export const EngagementCommentDashboard = () => {
+    const isLoggedIn = useAppSelector((state) => state.user.authentication.authenticated);
+    const isPreview = isLoggedIn;
+    const { savedEngagement } = useContext(ActionContext);
```

Comment:
if you only need the savedEngagement you don't need to pull the whole context from a different folder.

You can fetch the engagement in this file itself like it is being fetched in the action context. This way there is unneeded dependency between this file and a context for another component.

Reply:
or you can create a context for this component specifically.